### ISSUE TITLE
archivos json servidor y cliente

### ISSUE THREAD

**Issue by augustayanez** [IIC2233/Syllabus#507]
### Prerrequisitos
(Marcar colocando una `X` entre los corchetes los ítems que ya hiciste, así: `"[X]"`)

* [X] Leí las reglas del foro (https://github.com/IIC2233/Syllabus/wiki/3.-Reglas-del-foro-(issues)).
* [X] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/Syllabus/issues/).
* [X] Revisé el compilado de dudas de la tarea y no encontré una *issue* similar a la mía ([click aquí para ir al compilado](/IIC2233/Syllabus/issues/504)).
* [X] Mi duda no se trata sobre una librería, *built-in* o mala práctica, ya que eso se pregunta en la *issue* creada para ello ([click aquí para ir a esa _issue_](/IIC2233/Syllabus/issues/503)).
* [X] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al ayudante de Bienestar o al correo del curso (coordinacion.iic2233@gmail.com).
* [X] Utilizaré un título descriptivo y llenaré correctamente esta plantilla.
* [X] De ser necesario, colocaré código simple que permita explicar mi problema o duda, ya que compartir código de esta evaluación puede ser considerado una falta a la ética.

### Duda

En el enunciado dice que tanto el servidor como el cliente tendrán un archivo de formato json en sus carpetas con el host para instaciar un socket, estos archivos tenemos que crearlos nosotros?

**Comment by VAMarques** [comment]
Si, como el archivo JSON no se encuentra dentro de los que estan en las assets entregadas en la publicacion de la tarea, deberias crearlo tu mismo, por suerte su formato ya se dispone dentro del enunciado. El host recomendado, hasta donde yo se, es "localhost", pues eso permitira que corras el servidor y cliente desde tu propio computador.

**Comment by nicolettadegoye** [comment]
No entiendo lo del enunciado, se refiere que tiene que ser un archivo.py que se le importa json o es directamente un archivo.json?

**Comment by VAMarques** [comment]
> No entiendo lo del enunciado, se refiere que tiene que ser un archivo.py que se le importa json o es directamente un archivo.json?

Es directamente un archivo JSON, si los contenidos no han cambiado mucho creo que todavia ven la libreria json dentro del curso, entonces usas esta libreria para poder abrir los archivos json y tratarlos como diccionarios dentro de tu tarea.

**Comment by VAMarques** [comment]
> > No entiendo lo del enunciado, se refiere que tiene que ser un archivo.py que se le importa json o es directamente un archivo.json?
> 
> Es directamente un archivo JSON, si los contenidos no han cambiado mucho creo que todavia ven la libreria json dentro del curso, entonces usas esta libreria para poder abrir los archivos json y tratarlos como diccionarios dentro de tu tarea.

Efectivamente, lo revise en los contenidos y todavia se ve la libreria json, sin embargo no ven dump y load para json, los cuales funcionan igual que en pickle, asi que harias:

``````python
with open(path_de_mi_json, "rb") as archivo_json:
    # Puedes usar "rb" de forma analoga a como lo haces en pickle.load, pero he probado y tambien sirve hacerlo sin esto.
    # Lo que implica eso es que podrias usar "r" con un encoding, lo cual podria ser preferible si tu json tiene ciertos
    # caracteres extraneos no ascii, como los de latin-1, que me acuerdo habia un problema con eso en la tarea 3.
    mi_json = json.load(archivo_json)
``````

**Comment by nicolettadegoye** [comment]
En que parte de los contenidos esta?

**Comment by VAMarques** [comment]
> En que parte de los contenidos esta?

Es la semana 11, el notebook 1, de serializacion y bytes
